docs/workflow/testing/libraries/testing.md

```diff
@@ -101,4 +101,21 @@ dotnet build /t:Test /p:Outerloop=true
 
 ### Running tests on a different target framework
 
-Each test project can potentially have multiple target frameworks. There are some tests that might be OS-specific, or might be testing an API that is available only on some target frameworks, so the `TargetFrameworks` property specifies the valid target frameworks.
\ No newline at end of file
+Each test project can potentially have multiple target frameworks. There are some tests that might be OS-specific, or might be testing an API that is available only on some target frameworks, so the `TargetFrameworks` property specifies the valid target frameworks.
+
+### Running tests in custom compilation modes
+
+There are several custom compilation modes for tests. These are enabled by setting a switch during the config.
+
+| Mode | Description | Prerequisites
+| --- | --- | --- |
+| TestSingleFile | Test compilation using the single file compilation mode | libs+clr |
+| TestNativeAot | Test by compiling the test using NativeAOT | libs+clr.aot |
+| TestCrossgen2 | Test compilation of the test/libraries into R2R binaries | libs+clr |
+
+To run a test in specific mode, simply build the tests after building the prerequisite subsets, and specify the test mode on a command line such as:
+```cmd
```

Comment:
NIT: Try to keep lists and code examples surrounded by blank lines, as per the recommended Markdown guidelines.